File: dotnet10/Dotnet10/Dto/StoreFruitPriceDto.cs
using System.ComponentModel.DataAnnotations;

namespace Dotnet10.Dto;

/// <summary>DTO for a store-specific fruit price. Equivalent to StoreFruitPriceDTO record in the Quarkus project.</summary>
public record StoreFruitPriceDto(
    StoreDto Store,

    [property: Range(0, float.MaxValue, ErrorMessage = "Price must be >= 0")]
    float Price
);

File: quarkus3-virtual/src/main/java/org/acme/dto/StoreFruitPriceDTO.java
package org.acme.dto;

import java.math.BigDecimal;

public record StoreFruitPriceDTO(StoreDTO store, BigDecimal price) {
  public StoreFruitPriceDTO {
    if ((price != null) && (price.signum() < 0)) {
      throw new IllegalArgumentException("Price must be >= 0");
    }
  }
}
